Python/samples/house-energy/sim/house_simulator.py

```diff
@@ -0,0 +1,142 @@
+        
+    def update_Tset(self, Tset_new):
+        self.Tset = Tset_new # Update to new setpoint temperature
+
+    def update_hvacON(self, hvacONnew):
+        self.hvacON = hvacONnew # update to new hvacON
+
+    def update_occupancy(self, occupancy_new):
+        self.occupancy = occupancy_new # update to new occupancy
+      
+    def update_Tin(self):
+        """Update inside temperation.
+        Describes the inside temperature evolution as a function of all other variables. 
+        """
+        # Note timestep is converted to seconds here, in order to keep units consistent in SI for update.
+        self.Tin = self.Tin - (self.timestep/60) / self.C * (self.K * (self.Tin - self.Tout) + self.Qhvac * self.hvacON)
+        
+        self.__next__()
+        self.Tset_to_plot.append(self.Tset)
+        self.Tin_to_plot.append(self.Tin)
+        self.time_to_plot.append(self.iteration * 5)
+
+    def get_Power(self):
+        COP = 3
+        Power = self.Phvac * self.hvacON * COP 
+        return Power
+
+    def show(self):
+
+        self.ax.clear()
+        self.ax.plot(self.time_to_plot, self.Tin_to_plot, label='Tin')
+        self.ax.plot(self.time_to_plot, self.Tset_to_plot, label='Tset')
+        self.ax.set_xlabel('Time [min]')
+        self.ax.set_ylabel(r'Temperature [$^\circ$C]')
+        plt.legend()
+        plt.pause(np.finfo(np.float32).eps)
+        
+ # print the object nicely   
+    def __str__(self):
+        string_to_print = []
+        for key in self.__dict__:
+            string_to_print.append("{key}='{value}'".format(key=key, value=self.__dict__[key]))
+        return ', '.join(string_to_print)
+    
+    def __repr__(self):
+        return self.__str__() 
+
+    def __iter__(self):
+        self.iteration = 0
+        return self
+    
+    def __next__(self):
+        if self.iteration <= self.max_iterations - 5:
+            self.iteration += 1
+            self.update_Tset(self.Tset_schedule[self.iteration])
+            self.Tset1 = self.Tset_schedule[self.iteration+1]
+            self.Tset2 = self.Tset_schedule[self.iteration+2]
+            self.Tset3 = self.Tset_schedule[self.iteration+3]
+            self.update_Tout(self.Tout_schedule[self.iteration])
+            self.update_occupancy(self.occupancy_schedule[self.iteration])
+        else:
+            StopIteration
+
+
+def time_to_index(days, hours, timestep=5):
+    hours_index = int(hours * 60 / timestep)
+    days_index = int(days * 24 * 60 / timestep)
+    return hours_index + days_index
+
+if __name__ == '__main__':
+    import random
+    house = House()
+    
+    for episode in range(2):
+        house.setup_schedule(days=1,
+                             timestep=5,
+                             schedule_index=2,
+                             max_iterations= 288)
+        for i in range(288):
+            house.update_hvacON(random.randint(0, 1))
+            house.update_Tin()
+            #print(house.Tin)
```

Comment:
Is this meant to be here? Just curious because it's commented out. It would probably be best to remove it as it looks like it's used for debugging.

Reply:
Thank you @nmann4, uncommented and pushed. Merging